src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,112 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+
+class TC_MODE_BASE:
```

Comment:
Needs some doc comments. I would avoid using filler works like base/support/helper/core because files and classes named like that end up being a dumping ground of unrelated things.

Reply:
class naming should be `CamelCase`

Reply:
Changed the names format and filename.